Add Sumcoinindex rate source

schyczewski opened the pull request:
New rate source for Sumcoin. It is based on the cddash rate source and has been tested using the provided tools.

rpanak-generalbytes commented on server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/sumcoin/sources/sumcoinindex/SumcoinindexResponse.java:
```diff
@@ -0,0 +1,70 @@
+    private BigDecimal exch_rate_sell;
+    private BigDecimal exch_rate_buy;
+    private int error;
+    String error_msg;
```
`private` modifier missing?

schyczewski replied:
Yup, added in https://github.com/GENERALBYTESCOM/batm_public/pull/137/commits/4cdca4845ba2aef8dc91834f92968e1594d39613

rpanak-generalbytes commented on server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/sumcoin/sources/sumcoinindex/SumcoinindexResponse.java:
```diff
@@ -0,0 +1,70 @@
+    private int error;
+    String error_msg;
+    
+    public BigDecimal getexch_rate() {
```
Please follow Java camelCase naming conventions (throughout the whole file). You should be able to use the `@JsonProperty` annotation if JSON property doesn't match field name.

schyczewski replied:
Switched to camel case in https://github.com/GENERALBYTESCOM/batm_public/pull/137/commits/4cdca4845ba2aef8dc91834f92968e1594d39613

schyczewski commented:
I've implemented the requested changes, however I was unable to test them due to a build error that is seemingly unrelated, a failed test in CoinmarketcapAPITest (specifically an assertion at line 206). I reverted back to https://github.com/GENERALBYTESCOM/batm_public/pull/137/commits/2655986239465cc499fa2a32cce0761f78f86896 (a version that worked last I tested it) and ran into the same error.

schyczewski commented on server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/sumcoin/sources/sumcoinindex/CompatSSLSocketFactory.java:
```diff
@@ -0,0 +1,119 @@
+ * to be compatible with the TLS/SSL configuration of the server (TODO: insert
+ * name here).
+ */
+public class CompatSSLSocketFactory extends SSLSocketFactory {
```
Class removed.

schyczewski commented on server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/sumcoin/sources/sumcoinindex/SumcoinindexRateSource.java:
```diff
@@ -0,0 +1,156 @@
+        if (!(Currencies.USD.equalsIgnoreCase(fiatCurrency))) {
+            return null;
+        }
+        SumcoinindexResponse ticker = api.getTicker("rates","price2.json");
```
Changed in https://github.com/GENERALBYTESCOM/batm_public/pull/137/commits/2ca6bb41b4c21d7f10499086f811d50e684a1c84

schyczewski commented:
Disregard earlier comment about the CMC API thing, just saw the commit that fixed it. Anyway, the requested changes were implemented and tested.

rpanak-generalbytes commented on server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/sumcoin/sources/sumcoinindex/SumcoinindexRateSource.java:
```diff
@@ -0,0 +1,152 @@
+    public SumcoinindexRateSource() {
+	final ClientConfig config = new ClientConfig();
+        try {
+            SSLContext sslcontext=SSLContext.getInstance("TLS");
```
I think after your previous changes the `SSLContext` and `ClientConfig` is not needed anymore. You should also be able to remove the try-catch leaving the `api` assignment as the only statement in the constructor. This will simplify the constructor considerably.

schyczewski replied:
Done

rpanak-generalbytes commented:
Merged.
Thanks for the PR.
